### PR TITLE
Add a built in `PlaintextDecoder`

### LINKED ISSUE #2607: Missing PlaintextDecoder means users need to fall back to req.body to parse plain text http apis
**Is your feature request related to a problem? Please describe.**
As per https://discordapp.com/channels/431917998102675485/435928766812913664/833695763296813107 http apis which return content with Content-Type plain/text hit an Abort.httpDecoder error: Unsupported Media Type when decoding.

e.g.

```
return req.client.get(address).flatMapThrowing { res in
    try res.content.decode(UInt8.self)
}.map { n in
    /// do stuff
}
```

**Describe the solution you'd like**
I believe it would be in keeping with the rest of the API surface of vapor if this "just worked" (possibly with a requirement to specify the encoder)

I _think_ that means that I'd want a `PlaintextDecoder` (there's already a `PlaintextEncoder`) selected based on Content-Type plain/text

**Describe alternatives you've considered**
In my application I'm instead using `res.body!.getString(at: 0, length: res.body!.readableBytes)!` which feels strictly inferior (albeit hardly a lot of extra typing)

**Additional context**
None

### PR DESCRIPTION
**Pull request by dirtyhabits97**
Adds a new `PlaintextDecoder` to easily decode strings in requests and responses. `PlaintextDecoder` is the default decoder for `Content-Type: plain/text`. Resolves #2607

### CODE REVIEW

File: Tests/VaporTests/ContentTests.swift [hunk 1]
```diff
@@ -357,6 +357,23 @@ final class ContentTests: XCTestCase {
...
+
+        app.routes.get("plaintext") { (req) -> Response in
+            let res = Response()
+            try res.content.encode(data, as: .plainText)
+            return res
+        }
+
+        try app.testable().test(.GET, "/plaintext") { res in
+            XCTAssertEqual(res.status, .ok)
+            XCTAssertEqual(try res.content.decode(UInt8.self), 255)
```
Comment (0xTim) [inline Tests/VaporTests/ContentTests.swift:374]: Could you add a test that decodes a string as well? Otherwise LGTM

**Review by 0xTim — CHANGES_REQUESTED** [review]

File: Tests/VaporTests/ContentTests.swift (hunk 1 above)
Comment (dirtyhabits97) [reply Tests/VaporTests/ContentTests.swift:380]: Done. Also added a test case for empty string as the response's body.

**Review by 0xTim — APPROVED** [review]
Thanks! LGTM

**Comment by 0xTim** [comment]
Original comment:

> * Created PlaintextDecoder to easily decode raw strings
> * Registered PlaintextDecoder as the default decoder for Content-Type: plain/text
> This fixes #2607
